client/src/components/user_reviews/user_reviews.jsx

```diff
@@ -3,14 +3,66 @@ import UserReviewsEntry from '../user_reviews_entry/user_reviews_entry.jsx';
 const React = require('react');
 
 
-const UserReviews = (props) => {
-  return (
-    <ul>
-      {props.data.map((data) => {
-        return <UserReviewsEntry key={props.data.indexOf(data)} review={data} keywords={props.keywords} />;
-      })}
-    </ul>
-  );
-};
+class UserReviews extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      page: 1,
+    };
+    this.handleClick = this.handleClick.bind(this);
+    this.pageButton = this.pageButton.bind(this);
+  }
+
+  handleClick(e) {
+    this.setState({ page: parseInt(e.target.innerText) });
+  }
+
+  pageButton(pageNumber, currentPage, numberOfPages, pageOne) {
+    if (pageNumber === pageOne || pageNumber === numberOfPages) {
+      return true;
+    }
+    if (pageNumber < currentPage
+      && (currentPage - pageNumber) > 1
+      && (numberOfPages - pageNumber) !== 2) {
+      return false;
+    }
+    if (pageNumber > currentPage && (pageNumber - currentPage) > 1 && (pageNumber - 1) !== 2) {
+      return false;
+    }
+    return true;
+  }
+
+  render() {
+    return (
+      <div>
+        <ul>
+          {this.props.data.map((data) => {
+            if (data.pageNumber === this.state.page) {
+              return (
+                <UserReviewsEntry key={this.props.data.indexOf(data)} review={data} keywords={this.props.keywords} />
```

Comment:
Instead of using indexOf to find the index, you can set the index as a second parameter in your map method.

e.g. data.map((data, i) => ...